src/logic/calculate.js

```diff
@@ -0,0 +1,133 @@
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  if (isNumber(buttonName)) {
+    if (buttonName === '0' && obj.next === '0') {
+      return {};
+    }
+    // If there is an operation, update next
+    if (obj.operation) {
+      if (obj.next && obj.next !== '0') {
+        return { ...obj, next: obj.next + buttonName };
+      }
+      return { ...obj, next: buttonName };
+    }
+    // If there is no operation, update next and clear the value
+    if (obj.next && obj.next !== '0') {
+      return {
+        next: obj.next + buttonName,
+        total: null,
+      };
+    }
+    return {
+      next: buttonName,
+      total: null,
+    };
+  }
+
+  if (buttonName === '.') {
+    if (obj.next) {
+      if (obj.next.includes('.')) {
+        return { ...obj };
+      }
+      return { ...obj, next: `${obj.next}.` };
+    }
+    if (obj.operation) {
+      return { ...obj, next: '0.' };
+    }
+    if (obj.total) {
+      if (obj.total.includes('.')) {
+        return {};
+      }
+      return { ...obj, next: `${obj.total}.` };
+    }
+    return { ...obj, next: '0.' };
+  }
+
+  if (buttonName === '=') {
+    if (obj.next && obj.operation) {
+      return {
+        total: operate(obj.total, obj.next, obj.operation),
+        next: null,
+        operation: null,
+      };
+    }
+    // '=' with no operation, nothing to do
+    return {};
+  }
+
+  if (buttonName === '+/-') {
+    if (obj.next) {
+      return { ...obj, next: (-1 * parseFloat(obj.next)).toString() };
+    }
+    if (obj.total) {
+      return { ...obj, total: (-1 * parseFloat(obj.total)).toString() };
+    }
+    return {};
+  }
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
+
+  // User pressed an operation after pressing '='
+  if (!obj.next && obj.total && !obj.operation) {
+    return { ...obj, operation: buttonName };
+  }
+
+  // User pressed an operation button and there is an existing operation
+  if (obj.operation) {
+    if (obj.total && !obj.next) {
+      return { ...obj, operation: buttonName };
+    }
+
+    if (!obj.total) {
+      return { total: 0, operation: buttonName };
+    }
+
+    return {
+      total: operate(obj.total, obj.next, obj.operation),
+      next: null,
+      operation: buttonName,
+    };
+  }
+
+  // no operation yet, but the user typed one
+
+  // The user hasn't typed a number yet, just save the operation
+  if (!obj.next) {
+    return { operation: buttonName };
+  }
+
+  // save the operation and shift 'next' into 'total'
+  return {
+    total: obj.next,
+    next: null,
+    operation: buttonName,
+  };
+}
```

Comment:
- [ ] [REQUIRED] While performing some operations on the calculator, I discovered that it does not display as expected. Take a look at the 10 seconds video I attached. Kindly fix the display error as it is a requirement.


https://user-images.githubusercontent.com/58692590/199722462-8217fcb7-cadc-4035-92ea-48330655dfbe.mov

Reply:
Hi @Whoistolu. If that is the display error you demonstrated for me, it is already resolved. 8x9 = 72 and when you divide 72 by 5, you get 14 as a quotient and 2 as a remainder. Therefore, the functionality (modulus) works. After all, the functionality for the one you cited is implemented by **### operate.js** file, which is supposed to be dependable.
Thanks for the video demonstration that I can clearly see the bugs supposed to be resolved by me.

Reply:
> Hi @Whoistolu. If that is the display error you demonstrated for me, it is already resolved. 8x9 = 72 and when you divide 72 by 5, you get 14 as a quotient and 2 as a remainder. Therefore, the functionality (modulus) works. After all, the functionality for the one you cited is implemented by **_operate.js_** file, which is supposed to be dependable. Thanks for the video demonstration that I can clearly see the bugs supposed to be resolved by me.

- [ ] [NOTE] Hello next reviewer, all operations for this project works.
The changes should have been optional and not required. Thank you.